File: code/try-catch.java
//try catch format
int readNumber(BufferedReader reader) throws IOException {
    String line = null;
    try {
        line = reader.readLine();
        return Integer.parseInt(line);
    } catch (IOException e) {
        return 0;
    }finally {
        reader.close();
    }
}

//Unhandled Chcecked Exception
public Integer firstLetterOfYourName(String letter) throws IOException {
       if (!letter.matches("[a-zA-Z]+"))
            throw new IOException("ONE LETTER, please!");
   }

// You should add throws IOException or try catch 
   public void tryFirstLetterOfYourName() {
       firstLetterOfYourName("a"); // Unhandled Exception
   }

File: code/try-catch.kt
fun readNumber (reader :BufferedReader): Int?{
    try {
        val line = reader.readLine()
        return Integer.parseInt(line)
    }catch (e :NumberFormatException){
        return null
    }finally {
        reader.close()
    }
}

//try as expression
fun readNumber2 (reader : BufferedReader){
    //try just like if and when in kotlin it can be expression
    val number = try {
        val line = reader.readLine()
        Integer.parseInt(line)
    }catch (e :NumberFormatException){
        return
    }finally {
        reader.close()
    }
}

// All exceptions in Kotlin are unchecked exception.
fun firstLetterOfYourName(letter: Char){
    if (letter !in 'a'..'z' && letter !in 'A'..'Z')
        throw IOException("ONE LETTER, please!")
}

fun tryFirstLetterOfYourName() {
    firstLetterOfYourName('a')
}
